packages/@react-stately/list/src/useListState.ts

```diff
@@ -51,17 +51,48 @@ export function useListState<T extends object>(props: ListProps<T>): ListState<T
 
   let collection = useCollection(props, factory, context, [filter]);
 
+  let selectionManager = useMemo(() =>
+    new SelectionManager(collection, selectionState)
+    , [collection, selectionState]
+  );
+
   // Reset focused key if that item is deleted from the collection.
+  const cachedCollection = useRef(null);
   useEffect(() => {
     if (selectionState.focusedKey != null && !collection.getItem(selectionState.focusedKey)) {
-      selectionState.setFocusedKey(null);
+      const startItem = cachedCollection.current.getItem(selectionState.focusedKey);
+      const itemNodes = [...collection.getKeys()].map(
+        key => {
+          const itemNode = collection.getItem(key);
+          return itemNode.type === 'item' ? itemNode : null;
+        }
+      ).filter(node => node !== null);
+      let index = Math.min(startItem.index, itemNodes.length - 1);
+      let newNode:Node<T>;
+      while (index >= 0) {
+        if (!selectionManager.isDisabled(itemNodes[index].key)) {
```

Comment:
Same comment as in useGridState, prob should check for `selectionManager.disabledBehavior === 'all'` since focus behavior for disabled rows differs based on what is provided to ListView's `disabledBehavior` prop

Reply:
AFAIK, `selectionManager.isDisabled` checks `this.state.disabledBehavior === 'all'` at: https://github.com/adobe/react-spectrum/blob/93f49a443af04aba6ef2927f288eed340c788f5e/packages/%40react-stately/selection/src/SelectionManager.ts#L490-L492

Reply:
Ah, that is correct, this should be fine as is then